applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp

```diff
@@ -0,0 +1,272 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi,
+//                   Wijtze Pieter Kikstra
+//
+
+// Application includes
+#include "custom_constitutive/mohr_coulomb_with_tension_cutoff.h"
+#include "custom_constitutive/constitutive_law_dimension.h"
+#include "custom_utilities/math_utilities.h"
+#include "custom_utilities/stress_strain_utilities.h"
+#include "geo_mechanics_application_variables.h"
+#include "includes/serializer.h"
+
+namespace Kratos
+{
+MohrCoulombWithTensionCutOff::MohrCoulombWithTensionCutOff(std::unique_ptr<ConstitutiveLawDimension> pConstitutiveDimension)
+    : mpConstitutiveDimension(std::move(pConstitutiveDimension)),
+      mStressVector(ZeroVector(mpConstitutiveDimension->GetStrainSize())),
+      mStressVectorFinalized(ZeroVector(mpConstitutiveDimension->GetStrainSize())),
+      mStrainVectorFinalized(ZeroVector(mpConstitutiveDimension->GetStrainSize()))
+{
+}
+
+ConstitutiveLaw::Pointer MohrCoulombWithTensionCutOff::Clone() const
+{
+    auto p_result = std::make_shared<MohrCoulombWithTensionCutOff>(mpConstitutiveDimension->Clone());
+    p_result->mStressVector          = mStressVector;
+    p_result->mStressVectorFinalized = mStressVectorFinalized;
+    p_result->mStrainVectorFinalized = mStrainVectorFinalized;
+    return p_result;
```

Comment:
We also need to copy `mCoulombYieldSurface` and `mTensionCutoff`:
```suggestion
    p_result->mCoulombYieldSurface = mCoulombYieldSurface;
    p_result->mTensionCutoff = mTensionCutoff;
    return p_result;
```